DiscountDiscover/Managers/APIManager.m

```diff
@@ -53,6 +53,21 @@ - (void)fetchDeals:(double)radius completion:(void(^)(NSArray<Deal *> *deals, NS
     [task resume];
 }
 
+- (void)fetchNearbyDeal:(void (^)(Deal *deal, UIBackgroundFetchResult result))completionHandler {
+    [self fetchDeals:3 completion:^(NSArray<Deal *> *deals, NSError *error) {
+        if (error != nil) {
+            completionHandler(nil, UIBackgroundFetchResultFailed);
+        } else {
+            if (deals.count != 0) {
+                Deal *deal = deals[1];
```

Comment:
Why fetch 3 deals if you only use one? Also `deals[1]` might be out of bounds because you're only checking `deals.count != 0`, if `deals.count == 1` the app would crash here. You can use `deals.firstObject` as I mentioned in a previous comment, and should also fetch just one deal if that's all you need.